information/specification.md

```diff
@@ -0,0 +1,733 @@
+* [Background](#background)
+  * [What is Coin Selection?](#what-is-coin-selection)
+  * [Why is Coin Selection Non-Trivial?](#why-is-coin-selection-non-trivial)
+    * [Issues](#issues)
+    * [Goals](#goals)
+* [Definitions](#definitions)
+  * [Address](#address)
+  * [Coin Amount](#coin-amount)
+  * [Transaction Input](#transaction-input)
+  * [Transaction Output](#transaction-output)
+  * [UTxO Set](#utxo-set)
+* [Interface](#interface)
+  * [Parameters](#parameters)
+    * [Requested Output List](#requested-output-list)
+    * [Initial UTxO Set](#initial-utxo-set)
+    * [Maximum Input Count](#maximum-input-count)
+  * [Results](#results)
+    * [Coin Selection](#coin-selection)
+    * [Remaining UTxO Set](#remaining-utxo-set)
+  * [Properties](#properties)
+    * [Coverage of Payments](#coverage-of-payments)
+    * [Correctness of Change](#correctness-of-change)
+    * [Conservation of UTxO](#conservation-of-utxo)
+    * [Conservation of Outputs](#conservation-of-outputs)
+  * [Failure Modes](#failure-modes)
+    * [UTxO Balance Insufficient](#utxo-balance-insufficient)
+    * [UTxO Not Fragmented Enough](#utxo-not-fragmented-enough)
+    * [UTxO Fully Depleted](#utxo-fully-depleted)
+    * [Maximum Input Count Exceeded](#maximum-input-count-exceeded)
+* [Implementations](#implementations)
+  * [Largest-First](#largest-first)
+    * [Cardinality](#cardinality)
+    * [State](#state)
+      * [Available UTxO List](#available-utxo-list)
+      * [Unpaid Output List](#unpaid-output-list)
+      * [Accumulated Coin Selection](#accumulated-coin-selection)
+    * [Computation](#computation)
+    * [Termination](#termination)
+  * [Random-Improve](#random-improve)
+    * [Motivating Principles](#motivating-principles)
+      * [Principle 1: Dust Management](#principle-1-dust-management)
+      * [Principle 2: Change Management](#principle-2-change-management)
+      * [Principle 3: Performance Management](#principle-3-performance-management)
+    * [Cardinality](#cardinality-1)
+    * [State](#state-1)
+      * [Available UTxO Set](#available-utxo-set)
+      * [Accumulated Coin Selection](#accumulated-coin-selection-1)
+    * [Computation](#computation-1)
+      * [Phase 1: Random Selection](#phase-1-random-selection)
+      * [Phase 2: Improvement](#phase-2-improvement)
+    * [Termination](#termination-1)
+* [External Resources](#external-resources)
+  * [Self Organization in Coin Selection](#self-organization-in-coin-selection)
+
+# Purpose
+
+The purpose of this article is to describe, in _human-readable terms_, the coin
+selection algorithms used by Cardano Wallet and other parts of the Cardano
+ecosystem.
+
+# Background
+
+## What is Coin Selection?
+
+Coin selection is the process of choosing unspent coins from a wallet in order
+to pay money to one or more recipients.
+
+In the world of physical money, wallets hold value in the form of _coins and
+banknotes_.  When making a payment, we typically select a number of coins and
```

Comment:
The "typically" here makes me think of: "what is the alternative to selecting a number of coins and banknotes", and "is it true that I / one typically does this?", which I think distracts me from the point.

I wonder if it wouldn't be more effective to use a specific concrete example instead of having to _correctly_ generalise the typical scenario.

E.g, rather than
```
When making a payment [...] 
we typically select [...] 
it's often impossible to select the exact amount [...] "
in such cases we typically give the recipient more
```
what about something more like
```
Alice buys food [...] 
she selects [...] 
she cannot select the exact amount [...]
she pays more than"
```
?

Reply:
> I wonder if it wouldn't be more effective to use a specific concrete example instead of having to correctly generalise the typical scenario.

Perhaps we could provide both? Some people do seem to prefer general definitions, but it's clear that many people also like examples.

Reply:
Fixed in https://github.com/input-output-hk/cardano-coin-selection/pull/16/commits/638cd7749ea454037c0f835be180bbd654f73cdd.